tei_adapter/tei_updater/src/test_tei_updater.py

```diff
@@ -0,0 +1,148 @@
+# -*- encoding: utf-8 -*-
+
+
+import json
+import mock
+import os
+import pytest
+import requests
+
+from betamax import Betamax
+import boto3
+from botocore.exceptions import ClientError
+from moto import mock_s3
+
+from tei_updater import main
+from tei_updater import diff_trees
+
+pytest_plugins = "catalogue_aws_fixtures"
+
+with Betamax.configure() as config:
+    config.cassette_library_dir = "."
+
+
+@pytest.fixture
+def session():
+    session = requests.Session()
+    with Betamax(session) as vcr:
+        vcr.use_cassette("test_tei_updater")
+        yield session
+
+
+@mock_s3
+def test_tree_does_not_exist(
+    mock_sns_client, test_topic_arn, get_test_topic_messages, session
+):
+    bucket = "bukkit"
+    key = "tree.json"
+    mock_s3_client = boto3.client("s3", region_name="us-east-1")
+    mock_s3_client.create_bucket(Bucket=bucket)
+    with mock.patch.dict(
+        os.environ,
+        {
+            "TOPIC_ARN": test_topic_arn,
+            "BUCKET_NAME": bucket,
+            "TREE_FILE_KEY": key,
+            "GITHUB_API_URL": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/trees/master?recursive=true",
+        },
+    ):
+        main({}, s3_client=mock_s3_client, sns_client=mock_sns_client, session=session)
+    messages = get_test_topic_messages()
+    assert len(list(messages)) == 653
+    content_object = mock_s3_client.get_object(Bucket=bucket, Key=key)
+    body = content_object["Body"].read()
+    saved_tree = json.loads(body)
+    assert len(saved_tree.keys()) == 653
```

Comment:
```suggestion
        },
    ):
    
    main({}, s3_client=mock_s3_client, sns_client=mock_sns_client, session=session)
    
    messages = get_test_topic_messages()
    assert len(list(messages)) == 653
    
    content_object = mock_s3_client.get_object(Bucket=bucket, Key=key)
    body = content_object["Body"].read()
    saved_tree = json.loads(body)
    assert len(saved_tree.keys()) == 653
```

Reply:
You might also want to add some comments to these assertions to give a better idea of what's failed when it does? Maybe pytest also offers some nicer constructs to fail a test?

Reply:
yeah, will add a comment. Those numbers come from the recording of the github api. Not sure about more elegant assertions though